File: p007.c
#include <stdio.h>
#include <math.h>

int is_prime(int n) {
    if (n == 2) {
        return 1;
    } else if (n % 2 == 0) {
        return 0;
    }
    int to = (int)sqrt(n);
    int devide = 1;
    while (devide < to) {
        devide += 2;
        if (n % devide == 0) {
            return 0;
        }
    }
    return 1;
}


int main() {

    int found_cnt = 0, number = 1;
    while (found_cnt <= 10000) {
        number++;
        if (is_prime(number)) {
            found_cnt++;
        }
    }
    printf("%d\n", number);
    return;

}

File: p007.rb
#!/usr/bin/ruby

def check_is_prime(n)
  if n == 2
    return true
  elsif n % 2 == 0
    return false
  end
  devide = 1
  to = Math.sqrt(n).to_i
  while devide < to  do
    devide += 2
    if n % devide == 0 then
      return false
    end
  end
  return true
end

found_count = 0
number = 1
while found_count <= 10000 do
  number += 1
  if check_is_prime(number)
    found_count += 1
    #p number
  end
end
p number

